src/opnsense/service/templates/OPNsense/Shaper/dnctl.conf

```diff
@@ -7,8 +7,8 @@ pipe {{ pipe.number }} config bw {{ pipe.bandwidth }}{{ pipe.bandwidthMetric }}/
  if pipe.queue %} queue {{ pipe.queue }}{%
  if pipe.queueMetric != 'slots' %}{{pipe.queueMetric}}{% endif %}{% endif
  %}{% if pipe.buckets %} buckets {{ pipe.buckets }}{% endif
- %}{% if pipe.mask in ['src-ip', 'dst-ip'] %} mask {{ pipe.mask }} 0xffffffff {% endif
- %}{% if pipe.mask in ['src-ip6', 'dst-ip6'] %} mask {{ pipe.mask }} 128 {% endif %}{%
+ %}{% if pipe.mask in ['src-ip', 'dst-ip'] %} mask {{ pipe.mask }} /32 {% endif
+ %}{% if pipe.mask in ['src-ip6', 'dst-ip6'] %} mask {{ pipe.mask }} /128 {% endif %}{%
```

Comment:
does this mean the previous change (128 instead of /128) as actually incorrect?

Reply:
No, I don't believe so. dnctl accepts mask specifications in many different formats, and both produce the same results when I tested the dnctl command manually.

`128` triggers the following code path through [sbin/ipfw/dummynet.c:1556](https://github.com/freebsd/freebsd-src/blob/c3c816173d25fcb908a94c34b75fca98c5860f18/sbin/ipfw/dummynet.c#L1556), while `/128` triggers [sbin/ipfw/dummynet.c:1552](https://github.com/freebsd/freebsd-src/blob/c3c816173d25fcb908a94c34b75fca98c5860f18/sbin/ipfw/dummynet.c#L1552). At the end of the if/else block, `a` is always set to `128`, thus no net change.

Reply:
@tangrs sounds logical, thanks, I'll merge this one.